nft/simple_nft/tests/harness.rs

```diff
@@ -0,0 +1,393 @@
+    )
+    .await
+    .unwrap();
+
+    let deploy_wallet = Metadata {
+        nft: SimpleNFT::new(id.to_string(), wallet1.clone()),
+        wallet: wallet1,
+    };
+
+    let owner1 = Metadata {
+        nft: SimpleNFT::new(id.to_string(), wallet2.clone()),
+        wallet: wallet2,
+    };
+
+    let owner2 = Metadata {
+        nft: SimpleNFT::new(id.to_string(), wallet3.clone()),
+        wallet: wallet3,
+    };
+
+    (deploy_wallet, owner1, owner2)
+}
+
+async fn mint(
+    deploy_wallet: &Metadata,
+    owner1: &Metadata,
+) -> bool {
+    deploy_wallet
+        .nft
+        .mint(owner1.wallet.address())
+        .call()
+        .await
+        .unwrap()
+        .value
+}
+
+mod mint {
+    
+    use super::*;
+    
+    #[tokio::test]
+    async fn mints() {
+        let (deploy_wallet, owner1, _owner2) = setup().await;
+
+        assert!(
+            mint(
+                &deploy_wallet,
+                &owner1
+            )
+            .await
+        );
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn panics_when_minted_twice() {
+        let (deploy_wallet, owner1, _owner2) = setup().await;
+
+        mint(&deploy_wallet, &owner1).await;
+
+        assert!(
+            mint(
+                &deploy_wallet,
+                &owner1
+            )
+            .await
+        );   
+    }
+
+    // TODO: Test for wallet with 0 address
+    // #[tokio::test]
+    // #[should_panic]
+    // async fn panics_when_minted_to_zero_address() {
+    //     let (deploy_wallet, owner1, owner2) = setup().await;
+
+    //     let empty_address = Metadata {
+    //         nft: owner1.nft,
+    //         wallet: LocalWallet::, // Something needs to happen here
+    //     };
+
+    //     assert!(
+    //         mint(
+    //             &deploy_wallet,
+    //             &empty_address
+    //         )
+    //         .await
+    //     ); 
+    // }
+}
+
+mod burn {
+
+    use super::*;
+
+    #[tokio::test]
+    async fn burns() {
+        let (deploy_wallet, owner1, _owner2) = setup().await;
+
+        mint(&deploy_wallet, &owner1).await;
+
+        assert!(
+            owner1
+            .nft
+            .burn()
+            .call()
+            .await
+            .unwrap()
+            .value
+        );
+
+        assert_ne!(
+            owner1.nft.owner().call().await.unwrap().value,
+            owner1.wallet.address()
+        );
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn panics_when_burn_not_owner() {
+        let (deploy_wallet, owner1, owner2) = setup().await;
+
+        mint(&deploy_wallet, &owner1).await;
+
+        assert!(
+            owner2
+            .nft
+            .burn()
+            .call()
+            .await
+            .unwrap()
+            .value
+        );
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn panics_when_not_initalized() {
+        let (_deploy_wallet, owner1, _owner2) = setup().await;
+
+        assert!(
```

Comment:
no need to assert on a panic. the `unwrap()` will panic so the `.value` is not needed

Reply:
same in other places